subsys/bluetooth/host/hci_core.c

```diff
@@ -749,6 +754,34 @@ static int hci_le_read_remote_features(struct bt_conn *conn)
 	return 0;
 }
 
+int hci_read_remote_version(struct bt_conn *conn, bt_remote_version_cb *rv_cb)
+{
+	int err = k_sem_take(&remote_version_sem, HCI_CMD_TIMEOUT);
+	if (err != 0) {
+		return err;
+	} else {
+		struct bt_hci_cp_read_remote_version_info *cp;
+		struct net_buf *buf;
+
+		buf = bt_hci_cmd_create(BT_HCI_OP_READ_REMOTE_VERSION_INFO, sizeof(*cp));
+		if (!buf) {
+			return -ENOBUFS;
+		}
+
+		cp = net_buf_add(buf, sizeof(*cp));
+		cp->handle = sys_cpu_to_le16(conn->handle);
+		remote_version_cb = rv_cb;
```

Comment:
Move this assignment under the err handling of checking the command?  Then you don't need to unassign it in the if-block.